x-pack/plugins/alerts/server/health/get_state.test.ts

```diff
@@ -5,73 +5,187 @@
   });
 
-  it('should return an object with the "available" level and proper summary of "Alerting framework is healthy"', async () => {
-    mockTaskManager.get.mockReturnValue(
-      new Promise((_resolve, _reject) => {
-        return {
-          id: 'test',
-          attempts: 0,
-          status: TaskStatus.Running,
-          version: '123',
-          runAt: new Date(),
-          scheduledAt: new Date(),
-          startedAt: new Date(),
-          retryAt: new Date(Date.now() + 5 * 60 * 1000),
-          state: {
-            runs: 1,
-            health_status: HealthStatus.OK,
-          },
-          taskType: 'alerting:alerting_health_check',
-          params: {
-            alertId: '1',
-          },
-          ownerId: null,
-        };
+  it('should return unavailable status when health status is "error"', async () => {
+    const mockTaskManager = taskManagerMock.createStart();
+    mockTaskManager.get.mockResolvedValue(
+      getHealthCheckTask({
+        state: {
+          runs: 1,
+          health_status: HealthStatus.Error,
+        },
       })
     );
-    getHealthStatusStream(mockTaskManager).subscribe(
-      (val: { level: Readonly<unknown>; summary: string }) => {
-        expect(val.level).toBe(true);
+
+    const status = await getHealthServiceStatusWithRetryAndErrorHandling(
+      mockTaskManager
+    ).toPromise();
+
+    expect(status.level).toEqual(ServiceStatusLevels.unavailable);
+    expect(status.summary).toEqual('Alerting framework is unavailable');
+    expect(status.meta).toBeUndefined();
+  });
+
+  it('should retry on error and return healthy status if retry succeeds', async () => {
+    const retryDelay = 10;
+    const mockTaskManager = taskManagerMock.createStart();
+    mockTaskManager.get
+      .mockRejectedValueOnce(new Error('Failure'))
+      .mockResolvedValue(getHealthCheckTask());
+
+    getHealthServiceStatusWithRetryAndErrorHandling(mockTaskManager, retryDelay).subscribe(
+      (status) => {
+        expect(status.level).toEqual(ServiceStatusLevels.available);
+        expect(status.summary).toEqual('Alerting framework is available');
+      }
+    );
+
+    await tick();
+    jest.advanceTimersByTime(retryDelay * 2);
+  });
+
+  it('should retry on error and return unavailable status if retry fails', async () => {
+    const retryDelay = 10;
+    const err = new Error('Failure');
+    const mockTaskManager = taskManagerMock.createStart();
+    mockTaskManager.get.mockRejectedValue(err);
+
+    getHealthServiceStatusWithRetryAndErrorHandling(mockTaskManager, retryDelay).subscribe(
+      (status) => {
+        expect(status.level).toEqual(ServiceStatusLevels.unavailable);
+        expect(status.summary).toEqual('Alerting framework is unavailable');
+        expect(status.meta).toEqual({ error: err });
       }
     );
+
+    for (let i = 0; i < MAX_RETRY_ATTEMPTS + 1; i++) {
+      await tick();
+      jest.advanceTimersByTime(retryDelay);
+    }
```

Comment:
Could we add an assertion that `mockTaskManager.get` was actually called `MAX_RETRY_ATTEMPTS` times?

Otherwise, in theory, this test. won't catch anything if the stream never emits any values as all `expect`s are inside the subscription handler.
(this is from experience... I've missed a bug before because I made the exact same mistake) :)